frontend-react/src/hooks/network/ManagePublicKey/ManagePublicKeyHooks.ts

```diff
@@ -0,0 +1,183 @@
+import { useReducer } from "react";
+import pick from "lodash.pick";
+
+import {
+    PAYLOAD_MAX_BYTES,
+    PAYLOAD_MAX_KBYTES,
+} from "../../../utils/FileUtils";
+import { ContentType, Format } from "../../../utils/TemporarySettingsAPITypes";
+
+// Internal state for the hook.
+export interface ManagePublicKeyState {
+    fileInputResetValue: number;
+    fileContent: string;
+    contentType?: ContentType.PEM;
+    fileType?: Format.PEM;
+    fileName: string;
+    selectedSender: string;
+    reportId: string;
+    successTimestamp?: string;
+    localError: string;
+}
+
+export enum ManagePublicKeyActionType {
+    RESET = "RESET",
+    PREPARE_FOR_REQUEST = "PREPARE_FOR_REQUEST",
+    FILE_SELECTED = "FILE_SELECTED",
+    REQUEST_COMPLETE = "REQUEST_COMPLETE",
+}
+
+// TODO: Update with correct class when implementing saving of key
+export interface RequestCompletePayload {
+    response: {
+        id?: string;
+        timestamp?: string;
+    };
+}
+
+interface FileSelectedPayload {
+    file: File;
+}
+
+type ManagePublicKeyActionPayload =
+    | RequestCompletePayload
+    | FileSelectedPayload
+    | null;
+
+interface ManagePublicKeyAction {
+    type: ManagePublicKeyActionType;
+    payload?: ManagePublicKeyActionPayload; // reset actions will have no payload
+}
+
+type ManagePublicKeyReducer = (
+    state: ManagePublicKeyState,
+    action: ManagePublicKeyAction
+) => ManagePublicKeyState;
+
+export const INITIAL_STATE = {
+    fileInputResetValue: 0,
+    fileContent: "",
+    fileName: "",
+    selectedSender: "",
+    reportId: "",
+    successTimestamp: "",
+    localError: "",
+};
+
+function getInitialState(): ManagePublicKeyState {
+    return INITIAL_STATE;
+}
+
+// state for resetting / setting state at beginning of submission
+function getPreSubmitState(): Partial<ManagePublicKeyState> {
+    return {
+        ...pick(INITIAL_STATE, ["reportId", "successTimestamp", "localError"]),
+    };
+}
+
+// update state when file is selected in form
+function calculateFileSelectedState(
+    state: ManagePublicKeyState,
+    payload: FileSelectedPayload
+): Partial<ManagePublicKeyState> {
+    const { file } = payload;
+    try {
+        // look at the filename extension.
+        const fileNameArray = file.name.split(".");
+        const uploadFileExtension = fileNameArray[fileNameArray.length - 1];
+
+        if (uploadFileExtension.toUpperCase() !== Format.PEM) {
+            return {
+                ...state,
+                localError: "The file type must be of type .pem",
+            };
+        }
+
+        if (file.size > PAYLOAD_MAX_BYTES) {
```

Comment:
Might be worth breaking out the logic to check the file size into a separate function.  OR we could create a custom Form component that would do this automatically on file change.